pages/admin/reports/components/TotalEarnings.tsx

```diff
@@ -0,0 +1,260 @@
+import { Box, Text, useBreakpointValue } from '@chakra-ui/react';
+import {
+    BarChart,
+    ResponsiveContainer,
+    Bar,
+    XAxis,
+    YAxis,
+    Tooltip,
+    Legend,
+} from 'recharts';
+import { useEffect, useState } from 'react';
+import { getAppointments } from '@/services/AppointmentService';
+import { Appointment } from '@/interfaces/appointment/Appointment';
+import { splitDateTimeFromISO } from '@/utils/dateUtils';
+import {
+    TotalEarningsProps,
+    TreatmentStats,
+} from '@/interfaces/statistics/TotalEarnings';
+import parseTreatmentPrice from '@/utils/priceUtils';
+
+export default function TotalEarnings({
+    statistics,
+    dateRange,
+    selectedTreatmentId,
+    treatments,
+}: TotalEarningsProps) {
+    const [appointments, setAppointments] = useState<Appointment[]>([]);
+    const [filteredStats, setFilteredStats] = useState<TreatmentStats[]>([]);
+    const isMobile = useBreakpointValue({
+        base: true,
+        sm: true,
+        md: true,
+        lg: false,
+    });
+
+    const formatTreatmentName = (name: string) => {
+        if (!name) return '';
+        if (!isMobile) {
+            return name;
+        }
+        // If the name has spaces, use initials
+        const words = name.split(' ');
+        if (words.length > 1) {
+            return words.map((word) => word.charAt(0)).join('');
+        }
+        return name.length > 6 ? name.substring(0, 5) + '.' : name;
+    };
+
+    // Load appointments
+    useEffect(() => {
+        if (!statistics) {
+            setAppointments([]);
+            setFilteredStats([]);
+            return;
+        }
+        const fetchAppointments = async () => {
```

Comment:
Can add a flag this to avoid fetching if component has been already untmounted

```suggestion
  useEffect(() => {
    let isMounted = true;
    const fetchAppointments = async () => {
        try {
            const data = await getAppointments();
            if (isMounted) setAppointments(data);
        } catch (error) {
            console.error('Error al cargar citas:', error);
        }
    };
    fetchAppointments();
    return () => {
        isMounted = false;
    };
}, [statistics]);

```